service/src/main/kotlin/fi/espoo/evaka/occupancy/OccupancyController.kt

```diff
@@ -159,3 +241,60 @@ fun Database.Read.calculateOccupancyPeriodsGroupLevel(
         }
     }
 }
+
+private data class SpeculatedMaxOccupancies(
+    val current: OccupancyValues?,
+    val speculated: OccupancyValues?,
+)
+
+private fun calculateSpeculatedMaxOccupancies(
+    tx: Database.Read,
+    now: LocalDate,
+    unitId: DaycareId,
+    speculatedPlacements: List<Placement>,
+    from: LocalDate,
+    lengthsInMonths: List<Long>,
+): List<SpeculatedMaxOccupancies> {
+    val longestLength = lengthsInMonths.maxOrNull() ?: throw Error("TODO")
+    val longestPeriod = FiniteDateRange(from, from.plusMonths(longestLength).minusDays(1))
```

Comment:
I don't think subtracting a day is necessary, but I guess it's not clear what the 3/6 month maximum is supposed to mean anyway.